cantools/database/diagnostics/formats/cdd.py

```diff
@@ -98,7 +99,11 @@ def _load_data_types(ecu_doc):
                 LOGGER.debug("Ignoring unsupported attribute '%s'.", key)
 
         if ctype.attrib['bo'] == '21':
+            byte_order = 'big_endian'
+        elif ctype.attrib['bo'] == '12':
```

Comment:
doesn't this switch the default behavior? if yes, do you know why it worked for some file before?

Reply:
>doesn't this switch the default behavior? 

Yes.

> if yes, do you know why it worked for some file before?

It didn't really work before. The test cases passed because they made the same false assumption about endianness as the parser. I have now added files and test cases for both BigEndian and LittleEndian properties.